sqlmesh/core/reference.py

```diff
@@ -53,18 +53,66 @@ def name(self) -> str:
 
 class ReferenceGraph:
     def __init__(self, models: t.Iterable[Model]):
-        self._model_refs: t.DefaultDict[str, t.Dict[str, Reference]] = defaultdict(dict)
-        self._ref_models: t.DefaultDict[str, t.Dict[str, str]] = defaultdict(dict)
+        self._model_refs: t.Dict[str, t.Dict[str, Reference]] = {}
+        self._ref_models: t.Dict[str, t.Dict[str, str]] = {}
+        self._dim_models: t.Dict[str, t.Set[str]] = {}
 
         for model in models:
             self.add_model(model)
 
     def add_model(self, model: Model) -> None:
+        """Add a model and its references to the graph.
+
+        Args:
+            model: the model to add.
+        """
+        for column in model.columns_to_types or {}:
+            self._dim_models.setdefault(column, set())
+            self._dim_models[column].add(model.name)
+
         for ref in model.all_references:
+            self._model_refs.setdefault(model.name, {})
```

Comment:
Just curious why prefer this syntax over
```
self._model_refs[model.name] = {}
```
provided we don't use the return valuie.

Reply:
ah, it will only be set if the key is not there. Never mind